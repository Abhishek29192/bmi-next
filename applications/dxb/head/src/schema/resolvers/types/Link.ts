import type { Data as LinkData } from "../../../components/link/types";
import type { ParentPage } from "./Common";
import type { ContentfulFormSection } from "./FormSection";

export type ContentfulLink = Omit<
  LinkData,
  "linkedPage" | "dialogContent" | "id"
> & {
  sys: {
    id: string;
  };
  linkedPage: {
    __typename: string;
    sys: {
      id: string;
    };
    title: string;
    slug: string;
    parentPage: ParentPage | null;
  } | null;
  dialogContent: ContentfulFormSection | null;
};
