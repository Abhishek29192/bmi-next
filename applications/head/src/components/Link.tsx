import { graphql, Link } from "gatsby";
import { ClickableAction } from "@bmi/clickable";
import { IconName } from "./Icon";
import { Data as PageInfoData } from "./PageInfo";

export const getClickableActionFromUrl = (
  linkedPage: LinkData["linkedPage"],
  url: LinkData["url"],
  countryCode: string
): ClickableAction | undefined => {
  if (linkedPage && "slug" in linkedPage) {
    return {
      model: "routerLink",
      to: `/${countryCode}/${linkedPage.slug}`,
      linkComponent: Link
    };
  }

  if (url) {
    return {
      model: "htmlLink",
      href: url
    };
  }

  return undefined;
};

export const getCTA = (
  data:
    | {
        __typename: "ContentfulPromo";
        cta?: LinkData;
      }
    | Pick<PageInfoData, "__typename" | "slug">,
  countryCode: string,
  linkLabel?: string
) => {
  if (data.__typename === "ContentfulPromo") {
    if (!data.cta) {
      return null;
    }

    const { label, linkedPage, url } = data.cta;

    return {
      action: getClickableActionFromUrl(linkedPage, url, countryCode),
      label: label
    };
  }

  const { slug } = data;

  return {
    action: getClickableActionFromUrl({ slug: slug }, null, countryCode),
    // TODO: Use microcopy here
    label: linkLabel
  };
};

export type LinkData = {
  __typename: "ContentfulLink";
  id: string;
  label: string;
  icon: IconName | null;
  isLabelHidden: boolean | null;
  url: string | null;
  linkedPage: {
    // NOTE: null is for Homepage type
    slug: string | null;
  } | null;
};

export type NavigationItem = {
  __typename: "ContentfulNavigationItem";
  type: "Heading" | "Separator";
  value: string;
};

export type NavigationData = {
  __typename: "ContentfulNavigation";
  label: string | null;
  link: LinkData | null;
  links: (NavigationData | NavigationItem | LinkData)[];
};

export const query = graphql`
  fragment LinkFragment on ContentfulLink {
    __typename
    id
    label
    icon
    isLabelHidden
    url
    linkedPage {
      ... on ContentfulPage {
        slug
      }
    }
  }
`;
