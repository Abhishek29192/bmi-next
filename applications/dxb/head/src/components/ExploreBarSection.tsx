import ExploreBar from "@bmi-digital/components/explore-bar";
import Section from "@bmi-digital/components/section";
import { graphql } from "gatsby";
import React from "react";
import { devLog } from "../utils/devLog";
import { Data as LinkData, getClickableActionFromUrl } from "./Link";
import { useSiteContext } from "./Site";

export type Data = {
  __typename: "ContentfulNavigation";
  label: string | null;
  links: LinkData[];
};

const ExploreBarSection = ({ data }: { data: Data }) => {
  const { countryCode } = useSiteContext();
  const { label } = data;
  // Navigation is being used as the explore bar so bad values need filtering out.
  const links = data.links.filter((link) => {
    if (Object.keys(link).length) {
      return true;
    }
    devLog(
      `Only Links can be used if Navigation is being used as an Section. See Explore Bar labelled "${label}".`
    );
  });

  return (
    <Section>
      <ExploreBar
        heading={label}
        links={links.map(({ label, linkedPage, url, asset }) => ({
          label,
          action: getClickableActionFromUrl(
            linkedPage,
            url,
            countryCode,
            asset?.file?.url,
            label
          )
        }))}
      />
    </Section>
  );
};

export default ExploreBarSection;

export const query = graphql`
  fragment ExploreBarSectionFragment on ContentfulNavigation {
    label
    links {
      ...LinkFragment
    }
  }
`;
