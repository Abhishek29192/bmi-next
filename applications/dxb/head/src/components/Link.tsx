import uniqueId from "lodash-es/uniqueId";
import { Clickable, ClickableAction } from "@bmi-digital/components";
import { Dialog } from "@bmi-digital/components";
import { graphql, Link as GatsbyLink } from "gatsby";
import React, { useCallback, useContext, useMemo, useState } from "react";
import { Data as SimplePageData } from "../templates/simplePage/components/simple-page";
import { getPathWithCountryCode } from "../utils/path";
import { IconName } from "./Icon";
import { Data as PageInfoData } from "./PageInfo";
import { CalculatorContext } from "./PitchedRoofCalcualtor";
import { Data as PromoData } from "./Promo";
import { SectionData, sectionsMap } from "./Sections";
import { useSiteContext } from "./Site";
import styles from "./styles/Link.module.scss";
import { VisualiserContext } from "./Visualiser";

const checkUrlAction = (url: string): boolean => {
  const actionUrls = ["mailto:", "tel:", "callto:"];
  return actionUrls.some((actionUrl) => url.startsWith(actionUrl));
};
export enum DataTypeEnum {
  External = "External",
  Internal = "Internal",
  Asset = "Asset",
  Visualiser = "Visualiser",
  Calculator = "Calculator",
  Dialog = "Dialog",
  HubSpotCta = "HubSpot CTA"
}

export const isExternalUrl = (url: string): boolean => {
  try {
    const linkUrl = new URL(url);
    return linkUrl.host !== window.location.host;
  } catch (e) {
    return false;
  }
};

// TODO: This whole function needs refactoring
export const getClickableActionFromUrl = (
  linkedPage?: Data["linkedPage"],
  url?: Data["url"],
  countryCode?: string,
  assetUrl?: string,
  label?: string,
  type?: Data["type"],
  onClick?: (...args: any) => void,
  gtmData?: {
    id: string;
    label: string;
    action: string;
  }
): (ClickableAction & { "data-gtm"?: string }) | undefined => {
  if (type === DataTypeEnum.Visualiser) {
    const dataGtm = gtmData || {
      id: "cta-visualiser1",
      action: "visualiser",
      label
    };

    return {
      model: "default",
      onClick,
      "data-gtm": JSON.stringify(dataGtm)
    };
  }

  if (type === DataTypeEnum.Calculator) {
    const dataGtm = gtmData || {
      id: "cta-calculator1",
      action: "calculator",
      label
    };

    return {
      model: "default",
      onClick,
      "data-gtm": JSON.stringify(dataGtm)
    };
  }

  if (type === DataTypeEnum.Dialog) {
    const dataGtm = gtmData || { id: "cta-click1", action: type, label };
    return {
      model: "default",
      onClick,
      "data-gtm": JSON.stringify(dataGtm)
    };
  }

  if (assetUrl) {
    const dataGtm = gtmData || {
      id: "cta-click1",
      action: assetUrl,
      label
    };
    return {
      model: "download",
      href: assetUrl,
      "data-gtm": JSON.stringify(dataGtm)
    };
  }

  if (!countryCode && process.env.GATSBY_DONT_USE_COUNTRY_CODE !== "true") {
    return;
  }

  if (linkedPage && "path" in linkedPage) {
    const to = getPathWithCountryCode(countryCode, linkedPage.path).replace(
      /\/+/gi,
      "/"
    );
    const dataGtm = gtmData || { id: "cta-click1", action: to, label };

    return {
      model: "routerLink",
      to,
      linkComponent: GatsbyLink,
      "data-gtm": JSON.stringify(dataGtm)
    };
  }

  if (url) {
    const externalLinkProps = {
      // NOTE: External links should always open in a new tab.
      target: "_blank",
      rel: "noopener noreferrer"
    };
    const dataGtm = gtmData || { id: "cta-click1", action: url, label };

    return {
      model: "htmlLink",
      href: url,
      ...(checkUrlAction(url) || !isExternalUrl(url) ? {} : externalLinkProps),
      "data-gtm": JSON.stringify(dataGtm)
    };
  }
};

export const getCTA = (
  data:
    | Pick<PromoData, "cta">
    | Pick<PageInfoData, "path">
    | Pick<SimplePageData, "cta">,
  countryCode: string,
  linkLabel?: string
) => {
  if ("cta" in data) {
    if (!data.cta) {
      return null;
    }

    const { label, linkedPage, url, asset } = data.cta;

    return {
      action: getClickableActionFromUrl(
        linkedPage,
        url,
        countryCode,
        asset?.file?.url,
        label
      ),
      label: label
    };
  }

  const { path } = data;
  if (!path) return null;

  return {
    action: getClickableActionFromUrl(
      { path },
      null,
      countryCode,
      null,
      linkLabel
    ),
    label: linkLabel
  };
};

// TODO: HubSpot CTA script does not work on live. This is an intermediate
// step which may be acceptable for a final solution.
export const getLinkURL = (data: Data) =>
  data?.type === "HubSpot CTA" &&
  data?.hubSpotCTAID &&
  process.env.GATSBY_HUBSPOT_ID &&
  process.env.GATSBY_HUBSPOT_CTA_URL
    ? `${process.env.GATSBY_HUBSPOT_CTA_URL}${process.env.GATSBY_HUBSPOT_ID}/${data?.hubSpotCTAID}`
    : data?.url;

export type Data = {
  __typename: "ContentfulLink";
  id: string;
  label: string;
  icon: IconName | null;
  isLabelHidden: boolean | null;
  url: string | null;
  type: DataTypeEnum | null;
  parameters: { [key: string]: any } | null;
  dialogContent: SectionData | null;
  linkedPage: {
    // NOTE: null is for Homepage type
    path: string | null;
  } | null;
  asset?: {
    file: {
      url: string | null;
    };
  } | null;
  hubSpotCTAID: string | null;
};

export type NavigationItem = {
  __typename: "ContentfulNavigationItem";
  type: "Heading" | "Separator";
  value: string;
};

export type NavigationData = {
  __typename: "ContentfulNavigation";
  label: string | null;
  link: Data | null;
  promos?: PromoData[] | null;
  links: (NavigationData | NavigationItem | Data)[];
};

export const renderDialog = (
  data: Data,
  dialogIsOpen: boolean,
  handleDialogCloseClick: () => void
) => {
  if (!data?.dialogContent) {
    return;
  }

  const sectionId = `dialog-section-${uniqueId()}`;
  const Component: React.ElementType =
    sectionsMap[data.dialogContent.__typename];

  return (
    <Dialog open={dialogIsOpen} onCloseClick={handleDialogCloseClick}>
      <div className={styles["Link--dialog"]}>
        <Component data={data.dialogContent} id={sectionId} />
      </div>
    </Dialog>
  );
};
export const Link = ({
  children,
  component: Component = Clickable,
  data,
  onClick,
  ...rest
}: typeof Component & {
  children: React.ReactNode;
  component?: React.ElementType;
  data: Data;
  onClick?: (...args: any) => void;
}) => {
  const [dialogIsOpen, setDialogIsOpen] = useState(false);
  const { countryCode } = useSiteContext();
  const { open: openVisualiser } = useContext(VisualiserContext);
  const { open: openCalculator } = useContext(CalculatorContext);

  const handleOnClick = useCallback(
    (...args) => {
      onClick && onClick(...args);

      if (data?.type === "Visualiser" && openVisualiser) {
        openVisualiser(data?.parameters);
      } else if (data?.type === "Calculator" && openCalculator) {
        openCalculator(data?.parameters);
      } else if (data?.type === "Dialog") {
        setDialogIsOpen(true);
      }
    },
    [data?.parameters, data?.type, onClick]
  );

  const action = useMemo(
    () =>
      getClickableActionFromUrl(
        data?.linkedPage,
        getLinkURL(data),
        countryCode,
        data?.asset?.file?.url,
        data?.label,
        data?.type,
        handleOnClick
      ),
    [data, countryCode, handleOnClick]
  );

  const handleDialogCloseClick = useCallback(() => {
    setDialogIsOpen(false);
  }, []);

  const memoedRenderDialog = useMemo(
    () => renderDialog(data, dialogIsOpen, handleDialogCloseClick),
    [data?.dialogContent, dialogIsOpen]
  );

  return (
    <>
      <Component action={action} onClick={handleOnClick} {...rest}>
        {children}
      </Component>
      {data?.type === "Dialog" && data?.dialogContent && memoedRenderDialog}
    </>
  );
};

export default Link;

export const query = graphql`
  fragment LinkFragmentNonRecursive on ContentfulLink {
    __typename
    id
    label
    icon
    isLabelHidden
    url
    type
    linkedPage {
      ... on ContentfulHomePage {
        path
      }
      ... on ContentfulPage {
        path
      }
    }
    asset {
      ... on ContentfulAsset {
        file {
          ... on ContentfulAssetFile {
            url
          }
        }
      }
    }
    parameters {
      ...VisualiserFragment
    }
    hubSpotCTAID
  }
  fragment LinkFragment on ContentfulLink {
    ...LinkFragmentNonRecursive
    dialogContent {
      # Potential to add more sections here.
      ...FormSectionFragmentNonRecursive
    }
  }
`;
