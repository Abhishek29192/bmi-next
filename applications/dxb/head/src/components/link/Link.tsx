import Dialog from "@bmi-digital/components/dialog";
import { replaceSpaces, transformHyphens } from "@bmi-digital/components/utils";
import classnames from "classnames";
import NextLink from "next/link";
import uniqueId from "lodash-es/uniqueId";
import React, { Suspense, useCallback, useContext, useState } from "react";
import { getPathWithCountryCode } from "../../utils/path";
import { CalculatorContext } from "../PitchedRoofCalculator";
import { useSiteContext } from "../Site";
import { VisualiserContext } from "../Visualiser";
import { sectionsMap } from "./common";
import {
  classes,
  StyledAnchorLink,
  StyledButton,
  StyledLinkDialog
} from "./styles";
import { DataTypeEnum } from "./types";
import type {
  DialogProps,
  HubSpotProps,
  LinkProps,
  Props,
  ToolProps
} from "./types";

const ExternalLink = ({
  to,
  hasBrandColours,
  className,
  gtm,
  children
}: LinkProps) => {
  return (
    <StyledAnchorLink
      className={classnames(className, hasBrandColours && classes.branded)}
      href={to}
      external
      gtm={{
        id: "cta-click1",
        action: to,
        label: children,
        ...gtm
      }}
      data-testid={`external-link-${replaceSpaces(children)}`}
    >
      {transformHyphens(children)}
    </StyledAnchorLink>
  );
};

const InternalLink = ({
  to,
  hasBrandColours,
  className,
  gtm,
  children
}: LinkProps) => {
  const { countryCode } = useSiteContext();
  const url = getPathWithCountryCode(countryCode, to);

  return (
    <StyledAnchorLink
      className={classnames(className, hasBrandColours && classes.branded)}
      component={NextLink}
      href={url}
      gtm={{
        id: "cta-click1",
        action: url,
        label: children,
        ...gtm
      }}
      data-testid={`internal-link-${replaceSpaces(children)}`}
    >
      {transformHyphens(children)}
    </StyledAnchorLink>
  );
};

const AssetLink = ({
  to,
  hasBrandColours,
  className,
  gtm,
  children
}: LinkProps) => {
  return (
    <StyledAnchorLink
      className={classnames(className, hasBrandColours && classes.branded)}
      href={to}
      external
      download
      gtm={{
        id: "cta-click1",
        action: to,
        label: children,
        ...gtm
      }}
      data-testid={`asset-link-${replaceSpaces(children)}`}
    >
      {transformHyphens(children)}
    </StyledAnchorLink>
  );
};

const VisualiserButton = ({
  parameters,
  onClick,
  hasBrandColours,
  className,
  children,
  gtm
}: ToolProps) => {
  const { open } = useContext(VisualiserContext);

  const handleOnClick = useCallback(
    (...args: unknown[]) => {
      onClick && onClick(...args);

      open?.(parameters);
    },
    [onClick, open, parameters]
  );

  return (
    <StyledButton
      className={classnames(className, hasBrandColours && classes.branded)}
      onClick={handleOnClick}
      gtm={{
        id: "cta-visualiser1",
        action: "visualiser",
        label: children,
        ...gtm
      }}
      data-testid={`visualiser-link-${replaceSpaces(children)}`}
    >
      {transformHyphens(children)}
    </StyledButton>
  );
};

const CalculatorButton = ({
  parameters,
  onClick,
  hasBrandColours,
  className,
  children,
  gtm
}: ToolProps) => {
  const { open } = useContext(CalculatorContext);

  const handleOnClick = useCallback(
    (...args: unknown[]) => {
      onClick && onClick(...args);

      open?.(parameters);
    },
    [onClick, open, parameters]
  );

  return (
    <StyledButton
      className={classnames(className, hasBrandColours && classes.branded)}
      onClick={handleOnClick}
      gtm={{
        id: "cta-calculator1",
        action: "calculator",
        label: children,
        ...gtm
      }}
      data-testid={`calculator-link-${replaceSpaces(children)}`}
    >
      {transformHyphens(children)}
    </StyledButton>
  );
};

const DialogButton = ({
  dialogContent,
  onClick,
  hasBrandColours,
  className,
  children,
  gtm
}: DialogProps) => {
  const [dialogIsOpen, setDialogIsOpen] = useState(false);

  const handleOnClick = useCallback(
    (...args: unknown[]) => {
      onClick && onClick(...args);

      setDialogIsOpen(true);
    },
    [onClick, setDialogIsOpen]
  );

  const handleDialogCloseClick = useCallback(() => {
    setDialogIsOpen(false);
  }, []);

  if (!dialogContent) {
    return;
  }

  const Component: React.ElementType = sectionsMap[dialogContent.__typename];

  return (
    <>
      <StyledButton
        className={classnames(className, hasBrandColours && classes.branded)}
        onClick={handleOnClick}
        gtm={{
          id: "cta-click1",
          action: DataTypeEnum.Dialog,
          label: children,
          ...gtm
        }}
        data-testid={`dialog-link-${replaceSpaces(children)}`}
      >
        {transformHyphens(children)}
      </StyledButton>
      <Dialog
        open={dialogIsOpen}
        onCloseClick={handleDialogCloseClick}
        data-testid={`dialog-container-${replaceSpaces(children)}`}
      >
        <StyledLinkDialog>
          <Suspense fallback={<div>loading...</div>}>
            <Component
              data={dialogContent}
              id={`dialog-section-${uniqueId()}`}
              isDialog
            />
          </Suspense>
        </StyledLinkDialog>
      </Dialog>
    </>
  );
};

const HubSpotLink = ({
  hubSpotCtaId,
  hasBrandColours,
  className,
  gtm,
  children
}: HubSpotProps) => {
  const to = `${process.env.NEXT_PUBLIC_HUBSPOT_CTA_URL}${process.env.NEXT_PUBLIC_HUBSPOT_ID}/${hubSpotCtaId}`;
  return (
    <StyledAnchorLink
      className={classnames(className, hasBrandColours && classes.branded)}
      href={to}
      gtm={{
        id: "cta-click1",
        action: to,
        label: children,
        ...gtm
      }}
      data-testid={`hubspot-link-${replaceSpaces(children)}`}
    >
      {transformHyphens(children)}
    </StyledAnchorLink>
  );
};

const Link = (props: Props) => {
  switch (props.data.type) {
    case DataTypeEnum.External:
      return (
        <ExternalLink
          to={props.data.url}
          hasBrandColours={props.hasBrandColours}
          className={props.className}
          gtm={props.gtm}
        >
          {props.children}
        </ExternalLink>
      );
    case DataTypeEnum.Internal:
      return (
        <InternalLink
          to={props.data.linkedPage?.path}
          hasBrandColours={props.hasBrandColours}
          className={props.className}
          gtm={props.gtm}
        >
          {props.children}
        </InternalLink>
      );
    case DataTypeEnum.Asset:
      return (
        <AssetLink
          to={`https:${props.data.asset?.url}`}
          hasBrandColours={props.hasBrandColours}
          className={props.className}
          gtm={props.gtm}
        >
          {props.children}
        </AssetLink>
      );
    case DataTypeEnum.Visualiser:
      return (
        <VisualiserButton
          onClick={props.onClick}
          parameters={props.data.parameters}
          hasBrandColours={props.hasBrandColours}
          className={props.className}
          gtm={props.gtm}
        >
          {props.children}
        </VisualiserButton>
      );
    case DataTypeEnum.Calculator:
      return (
        <CalculatorButton
          onClick={props.onClick}
          parameters={props.data.parameters}
          hasBrandColours={props.hasBrandColours}
          className={props.className}
          gtm={props.gtm}
        >
          {props.children}
        </CalculatorButton>
      );
    case DataTypeEnum.Dialog:
      return (
        <DialogButton
          onClick={props.onClick}
          dialogContent={props.data.dialogContent}
          hasBrandColours={props.hasBrandColours}
          className={props.className}
          gtm={props.gtm}
        >
          {props.children}
        </DialogButton>
      );
    case DataTypeEnum.HubSpotCta:
      return (
        <HubSpotLink
          hubSpotCtaId={props.data.hubSpotCTAID}
          hasBrandColours={props.hasBrandColours}
          className={props.className}
          gtm={props.gtm}
        >
          {props.children}
        </HubSpotLink>
      );
  }
};

export default Link;
