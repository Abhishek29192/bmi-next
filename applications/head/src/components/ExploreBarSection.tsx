import ExploreBar from "@bmi/explore-bar";
import Section from "@bmi/section";
import { graphql } from "gatsby";
import React, { useContext } from "react";
import { getClickableActionFromUrl, LinkData } from "./Link";
import { SiteContext } from "./Site";

export type Data = {
  __typename: "ContentfulNavigation";
  label: string | null;
  links: LinkData[];
};

const ExploreBarSection = ({ data }: { data: Data }) => {
  const { countryCode } = useContext(SiteContext);
  const { label, links } = data;

  return (
    <Section>
      <ExploreBar
        heading={label}
        links={links.map(({ label, linkedPage, url, asset }) => ({
          label,
          action: getClickableActionFromUrl(
            linkedPage,
            url,
            countryCode,
            asset?.file?.url,
            label
          )
        }))}
      />
    </Section>
  );
};

export default ExploreBarSection;

export const query = graphql`
  fragment ExploreBarSectionFragment on ContentfulNavigation {
    label
    links {
      ...LinkFragment
    }
  }
`;
