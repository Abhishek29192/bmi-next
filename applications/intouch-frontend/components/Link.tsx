import React from "react";
import Link from "next/link";
import Icon from "@bmi/icon";
import styles from "./styles/Sidebar.module.scss";

type Props = {
  href?: string;
  icon: React.FC; // TODO: convert into an SVG component
  label: string;
};

const SelectedLink = ({ href, icon, label }: Props) => {
  return (
    <Link href={href}>
      <a className={styles.sidebarLink}>
        <Icon source={icon} color="primary" style={{ fontSize: 24 }} />
        {label}
      </a>
    </Link>
  );
};

export default SelectedLink;
