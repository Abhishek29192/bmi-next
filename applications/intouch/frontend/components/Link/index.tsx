import React from "react";
import { default as NextLink } from "next/link";

interface LinkProps extends React.AnchorHTMLAttributes<any> {
  isExternal?: boolean;
  children: React.ReactNode;
}

export const Link = ({ children, href, isExternal, ...rest }: LinkProps) => {
  if (isExternal) {
    return (
      <a href={href} target="_blank" rel="noopener noreferrer" {...rest}>
        {children}
      </a>
    );
  }
  return (
    <NextLink href={href} {...rest}>
      {children}
    </NextLink>
  );
};
