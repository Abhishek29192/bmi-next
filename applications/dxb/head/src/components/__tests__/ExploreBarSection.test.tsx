import React from "react";
import { render } from "@testing-library/react";
import ExploreBarSection, { Data } from "../ExploreBarSection";
import { DataTypeEnum } from "../Link";

describe("ExploreBarSection component", () => {
  it("renders correctly", () => {
    const data: Data = {
      __typename: "ContentfulNavigation",
      label: "Lorem ipsum",
      links: [
        {
          __typename: "ContentfulLink",
          id: "link",
          label: "String",
          icon: null,
          isLabelHidden: false,
          url: "link-to-page",
          linkedPage: null,
          type: DataTypeEnum.Internal,
          parameters: null,
          dialogContent: null,
          hubSpotCTAID: null
        },
        {
          __typename: "ContentfulLink",
          id: "link2",
          label: "Another string",
          icon: null,
          isLabelHidden: false,
          url: "link-to-page-2",
          linkedPage: null,
          type: DataTypeEnum.Internal,
          parameters: null,
          dialogContent: null,
          hubSpotCTAID: null
        },
        // @ts-ignore This is how a nested Navigation comes through from Contentful
        {}
      ]
    };

    const { container } = render(<ExploreBarSection data={data} />);

    expect(container).toMatchSnapshot();
  });
});
