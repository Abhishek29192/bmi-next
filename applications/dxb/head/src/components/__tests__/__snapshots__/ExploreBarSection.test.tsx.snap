// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExploreBarSection component renders correctly 1`] = `
<div
  class="Section"
>
  <div
    class="Container"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl"
    >
      <nav
        aria-labelledby="explore-bar"
        class="ExploreBar"
      >
        <h2
          class="MuiTypography-root heading Typography MuiTypography-h6"
          id="explore-bar"
        >
          Lorem ipsum
        </h2>
        <button
          class="MuiButtonBase-root MuiButton-root buttonStyles-root-1 MuiButton-outlined link MuiButton-outlinedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label buttonStyles-label-4"
          >
            String
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root buttonStyles-root-1 MuiButton-outlined link MuiButton-outlinedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label buttonStyles-label-4"
          >
            Another string
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </nav>
    </div>
  </div>
</div>
`;
